code/datums/uplink_items/uplink_general.dm

```diff
@@ -22,27 +20,51 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 				uplink_items[I.category] = list()
 
 			uplink_items[I.category] += I
-			if(I.limited_stock < 0 && I.can_discount && I.item && I.cost > 5)
+
+			//Add items to discount pool, checking job, species, and hijacker status
+			var/job_check = FALSE
+			var/species_check = FALSE
+			var/hijacker_check = FALSE
+			if(I.job) //If your job does not match, no discount
+				for(var/job in I.job)
+					if(job == user.mind.assigned_role)
+						job_check = TRUE
+			else {
+				job_check = TRUE
+			}
```

Comment:
```suggestion
			else
				job_check = TRUE
```
we don't use curly brackets here to wrap logic blocks

Reply:
Whoop. Sorry about that. That's my actual job leaking through.